### ISSUE TITLE
Failed to discover path of nested stack url

### ISSUE THREAD

**Issue by dynaknot** [aws-quickstart/taskcat#446]
When I test a master stack the Tackcat cannot resolve the url of the nested stack stored on S3 bucket.

**Steps to reproduce the behavior:**
Running `taskcat test run` command with the following `.taskcat.yml`: 

```
project:
  name: taskcat-scenario1-main
  owner: myemail
  regions:
    - us-east-1
tests:
  taskcat-scenario1-main:
    parameters:
      StackName: TestStack
      vAvailabilityZone: '$[taskcat_genaz_1]'
      vVPCCIDR: '10.0.0.0/16'
      vVPCTenancy: 'default'
      vVpcEnableDnsHostnames: 'true'
      vVpcEnableDnsSupport: 'true'
      vPublicSubnet1ACIDR: '10.0.100.0/24'
      vPublicSubnetTag1A: 'Network=Public'
      vPublicSubnet2ACIDR: '10.0.200.0/24'
      vPublicSubnetTag2A: 'Network=Public'
      vBastionInstanceType: 't2.micro'
      vEC2KeyName: 'mykeyname'
      vSSHLocation: '127.0.0.1/32'
      vOperatorAlertEmail: 'myemail'
      vLampInstanceType: 't2.micro'
    template: ./master-stack.yaml
```

In the master stack there is a nested stack resource:
```
Resource:
  VPCStack:
    Type: 'AWS::CloudFormation::Stack'
    Properties:
      TemplateURL: https://mybucketname.s3.amazonaws.com/mybucketfoldername/nested-stack-network.yaml
      Parameters:
        vAvailabilityZone: !Ref vAvailabilityZone
        vVPCCIDR: !Ref vVPCCIDR
        vPublicSubnet1ACIDR: !Ref vPublicSubnet1ACIDR
        vPublicSubnet2ACIDR: !Ref vPublicSubnet2ACIDR
```

Receiving the following message:
```
[WARN   ] : Failed to discover path for https://mybucketname.s3.amazonaws.com/mybucketfoldername/nested-stack-network.yaml, path /home/ec2-user/environment/cloud9foldername/mybucketfoldername/nested-stack-network.yaml does not exist
```

The Taskcat continues testing the rest of the master template then stops with fail.

**On the Cloud9 IDE I use the following directory and files:**
```
/home/ec2-user/environment/cloud9foldername/
.tackcat.yml
master-stack.yaml
```

**Control testing:**
I tried to run this master stack on the Cloudformation UI resulting the master stack and the nested stack has been created successful.

The S3 bucket is publicly accessible bucket and from a browser using the S3 url I can download the nested stack as a file.

**Expected behavior**
When I test a master stack the tackcat should resolve the s3 https url of the nested stack stored on a S3 bucket and continue running the test of the nested stack.

**Screenshots**
N/A

**Version**
 - Taskcat Version 0.9.8
 - Python Version: 3.6.8

1. Are you testing a *QuickStart* or *Custom template*? 
it is a custom template.

2. Attach or link a copy of the template if possible
N/A

3. Provide the parameters that you passed.
for the nested stack url I don't use parameters but I plan to use parameters constructing the url later.

4. How did you install taskcat? (docker or pip3)
I use AWS Cloud9 IDE and I followed the taskcat installation instruction `pip3 install taskcat`

5. Are you using a *profile*, *an instance role* or  *access keys* to run taskcat?
I use normal user profile in Cloud9 IDE.

6. Is your AWS environment configured via `aws configure`?
Yes this IDE is configured by AWS. I installed `phyton3` and `taskcat` only.

**Additional context**
This Cloud9 EDI has AWS generated SecurityGroup allowing: 

outbound rule: all ports, all protocol, destination 0.0.0.0/0,
inbound rule: 22 port, tcp protocol, source: cloud9 IPs.

From Cloud9 IDE I can ping more website's domain name successful.

**Comment by dynaknot** [comment]
After 3 weeks of waiting this case was also raised on AWS Support. After examining the case together with the AWS Support it seems to be a false message. They will ask Taskcat dev team to fix this false error message in this general case. Please be so kind to turn to AWS Support for ticket details.

**Comment by avattathil** [comment]
@gargana the nested stack linting logic should handle this use-case correct ?

**Comment by gargana** [comment]
@avattathil I am working to include the improved local detection logic used in our custom linting rules.
I am planning to add S3 support along with this too.

**Comment by andrew-glenn** [comment]
@dynaknot We added enhanced template URL support in #499, and was released with `0.9.17`; if you continue to run into issues, please reopen.
